File: practica2_839756_842236/c++/instruccionMul.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionMul.h"

void InstruccionMul::ejecutar(stack<int> *s)
{
    int x, y;
    x = s->top();
    s->pop();
    y = s->top();
    s->pop();
    s->push(y * x);
}

void InstruccionMul::listar()
{
    cout << "mul" << endl;
}

File: practica2_839756_842236/java/InstruccionMul.java
import java.util.Stack;

public class InstruccionMul extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        int x = s.pop();
        int y = s.pop();
        s.push(y * x);
    }

    @Override
    public void listar() {
        System.out.println("mul");
    }
}
